Bugfix/unique player

komailm opened the pull request.

komailm committed:
fix: user can be selected only once

komailm committed:
fix: user can be selected only once

komailm committed:
fix: user can be selected only once

geninOne commented on src/app/shared/components/add-match/select-players/select-players.component.ts:
```diff
@@ -19,4 +21,98 @@ export class SelectPlayersComponent {
+  touched: boolean[] = [false, false, false, false];
+
+  constructor(userService: UsersService, matchesService: MatchesService) {
+    userService.users$.subscribe((data: IUser[]) => {
```
Mostly we use ngOnInit for all the initialization/declaration and avoid stuff to work in the constructor. The constructor should only be used to initialize class members but shouldn't do actual "work".

So you should use constructor() to setup Dependency Injection and not much else. ngOnInit() is better place to "start" - it's where/when components' bindings are resolved.

geninOne commented on src/app/shared/components/add-match/select-players/select-players.component.ts:
```diff
@@ -19,4 +21,98 @@ export class SelectPlayersComponent {
+    });
+  }
+
+  getPlayersList(selId: string) {
```
The type should be a number. So you don't need the transformation in the function. Because in the HTML file you do a calculation, this should transform automatic your variable into a number.

komailm committed:
fix: user can be selected only once

moved subscriptions from constructor to ngOnInit,
changed type of parameter for getPlayersList,
added ngOnDestroy to unsubscribe
